### PR TITLE
Logic was implemented to update and remove users, tweets, and Bug fixed in ValidateUser inside 'middleware/validator.js' file

### PR DESCRIPTION
**Pull request by cags84**
- The schema was modified to receive the role parameter, this can be administrator 'admin' or registered.
- The authorization file was modified exactly the 'user Authorization' method to implement the required logic.
- For everything to work we have to create a service in 'api / services' with the name of userService.js where we place the logic to know if the user is an administrator or not.
- In userService.js we import the User model. and we consult the database with the id and with the condition if it is admin.
- functionality is implemented for you to save the role.
- Other minnor changes.

- Bug -> This error occurs when not all the required data is sent in the path 'api/users', by post method, then it generated an unexpected error, this bug was on line 84.

This error occurs when not all the required data is sent in the path 'api/users', by post method, then it generated an unexpected error, this bug was on line 84.

See error image.

After changing the line, this is the result.

### CODE REVIEW

**Comment by cags84** [comment]
fix delete method in 'users / controllers' and update logic to implement delete method `commit 1ca89ad`

    - Updated locate file.
    - Other minnor changes.

**Comment by cags84** [comment]
Functionality was implemented to delete tweet (commit 3dbbc50)

- functionality was implemented to delete tweet, only to those who create it or to administrator users.

File: api/middleware/authorization.js [hunk 1]
```diff
@@ -1,11 +1,29 @@
...
+const { locale } = require("../../locale");
+const { isAdmin, findUserWithUsername } = require("../services/userService");
+const { findUserByTweetId } = require("../services/tweetService");
 
-const usersAuthorization = (req, res, next) => {
-  next();
+const usersAuthorization = async (req, res, next) => {
+  const id = await findUserWithUsername(req.body.username);
+  const { userId } = req.body;
+  const isAdminTemp = await isAdmin(userId);
```
Comment (jestrade) [inline api/middleware/authorization.js:8]: puedes implementar el método isAdmin para que no sea una promesa y no requieras  el llamado de await, así lo puedes usar en el if (... && isAdmin).

File: api/middleware/authorization.js [hunk 2]
```diff
@@ -1,11 +1,29 @@
...
+
+  if (userId == id || isAdminTemp === true) {
```
Comment (jestrade) [inline api/middleware/authorization.js:10]: Recuerda que cuando validas === true puedes omitir esa parte de la pregunta

File: api/middleware/authorization.js [hunk 3]
```diff
@@ -1,11 +1,29 @@
...
+    next();
+  } else {
+    res.status(403).send("Operation not allowed!");
```
Comment (jestrade) [inline api/middleware/authorization.js:13]: usar location

File: api/middleware/authorization.js [hunk 4]
```diff
@@ -1,11 +1,29 @@
...
-  next();
+const tweetsAuthorization = async (req, res, next) => {
+  const id = await findUserByTweetId(req.body.tweetId);
+  const { userId } = req.body;
+  const isAdminTemp = await isAdmin(userId);
+
+  if (userId == id || isAdminTemp === true) {
+    next();
+  } else {
+    res.status(403).send("Operation not allowed!");
```
Comment (jestrade) [inline api/middleware/authorization.js:25]: usar location

File: api/middleware/authorization.js [hunk 5]
```diff
@@ -1,11 +1,29 @@
...
 };
 
-const tweetsAuthorization = (req, res, next) => {
-  next();
+const tweetsAuthorization = async (req, res, next) => {
+  const id = await findUserByTweetId(req.body.tweetId);
+  const { userId } = req.body;
+  const isAdminTemp = await isAdmin(userId);
+
+  if (userId == id || isAdminTemp === true) {
```
Comment (jestrade) [inline api/middleware/authorization.js:22]: lo mismo que arriba

File: api/middleware/validator.js [hunk 6]
```diff
@@ -4,10 +4,10 @@ const validateTweet = (req, res, next) => {
 
   if (content) {
     if (content.length > 280) {
-      errors.push('max characters exceded');
+      errors.push("max characters exceded");
```
Comment (jestrade) [inline api/middleware/validator.js:7]: locale

**Review by jestrade — CHANGES_REQUESTED** [review]
some minor comments

**Pull request closed without merging**